streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -611,12 +615,12 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
     static <K1, V1> String createRepartitionedSource(final InternalStreamsBuilder builder,
                                                      final Serde<K1> keySerde,
                                                      final Serde<V1> valSerde,
-                                                     final String topicNamePrefix,
+                                                     final String repartitionTopicNamePrefix,
                                                      final String name,
                                                      final OptimizableRepartitionNode.OptimizableRepartitionNodeBuilder<K1, V1> optimizableRepartitionNodeBuilder) {
 
-        final String baseName = topicNamePrefix != null ? topicNamePrefix : name;
-        final String repartitionTopic = baseName + REPARTITION_TOPIC_SUFFIX;
+        final String repartitionTopicBaseName = repartitionTopicNamePrefix != null ? repartitionTopicNamePrefix : name;
+        final String repartitionTopic = repartitionTopicBaseName.endsWith(REPARTITION_TOPIC_SUFFIX) ? repartitionTopicBaseName  : repartitionTopicBaseName + REPARTITION_TOPIC_SUFFIX;
```

Comment:
What's the code path that leads to `repartitionTopicBaseName.endsWith(REPARTITION_TOPIC_SUFFIX)`? I couldn't find it.

Reply:
It comes from the `repartitionTopicNamePrefix` passed as a parameter when calling `createRepartitionedSource`.

Since we may get an existing repartition topic name, we don't want to append `-repartition` at the end and change it.  

For example, we may get `KSTREAM-AGGREGATE-STATE-STORE-0000000005-repartition` for a repartition topic name resulting from an optimization operation and appending another `-repartition` would break the topology.  

As the java docs and our other docs state, we append as single `-repartition` to the repartition topic name we wouldn't want to double append.  But I'll update this with a comment explaining why we do this check.

Reply:
I agree we shouldn't append multiple suffixes. It just wasn't clear where the pre-suffixed string could come from. Thanks for the clarification!

Reply:
Cannot follow here... Do you aim for existing topologies with generated names, and user update code to "pin" names? For this case, user would pass it name, without `-repartition` suffix? User, would also need to drop `<application.id>` prefix in the name she passed to `Grouped`.

Reply:
I cannot follow here too.. the `createRepartitionedSource` should always be called before the optimization kicks in, so the passed in name should always be the raw names right?